### PR TITLE
HSEARCH-1824 Have the new state of a MutableSearchFactoryState not nuke ...

### PR DESCRIPTION
**Pull request by Sanne**
...the previously defined Filter(s) and Analyzer(s)

 - https://hibernate.atlassian.net/browse/HSEARCH-1824

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/impl/MutableSearchFactoryState.java [hunk 1]
```diff
@@ -41,9 +42,9 @@
 	private IndexingMode indexingMode;
 	private Worker worker;
 	private BackendQueueProcessor backendQueueProcessor;
-	private Map<String, FilterDef> filterDefinitions;
+	private Map<String, FilterDef> filterDefinitions = new ConcurrentHashMap<>();
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/search/engine/impl/MutableSearchFactoryState.java]: I would declare all the map members of type `ConcurrentMap` to make their nature apparent. Callers I would just let pass plain maps whose contents I'd put to the internally used concurrent maps. I.e. would not rely on the fact that a concurrent map is passed in (e.g. see factoryState.setDocumentBuildersIndexedEntities(...) further below in this PR).

File: engine/src/test/java/org/hibernate/search/test/configuration/mutablefactory/DefinitionsOnHotRebootTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.hibernate.search.testsupport.setup.SearchConfigurationForTest;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Makes sure that Filter and Analyzer definitions survive a hot reboot
+ */
+@TestForIssue(jiraKey = "HSEARCH-1230")
```
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/search/test/configuration/mutablefactory/DefinitionsOnHotRebootTest.java]: s/1230/1824/?

**Comment by gunnarmorling** [comment]
@Sanne Looks good to me apart from thos two remarks inline.

File: engine/src/test/java/org/hibernate/search/test/configuration/mutablefactory/DefinitionsOnHotRebootTest.java (hunk 2 above)
Comment (Sanne) [inline engine/src/test/java/org/hibernate/search/test/configuration/mutablefactory/DefinitionsOnHotRebootTest.java]: doh :) thanks!

File: engine/src/main/java/org/hibernate/search/engine/impl/MutableSearchFactoryState.java (hunk 1 above)
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/engine/impl/MutableSearchFactoryState.java]: I agree but I'd rather keep that out of this specific change. It has the potential to go viral and get me to refactor all of it, which I should definitely do but not in the scope of this micro fix which we need ASAP.

**Comment by Sanne** [comment]
merged

**Pull request closed without merging**